### PR TITLE
Add user requested by to the evidence request and show it in the UI

### PR DESCRIPTION
**Pull request by bogdan-zaharia**
- Add `userRequestedBy` to the evidence request
- Show `userRequestedBy` on the "Requests" and "Residents" pages
- Add cypress test

### CODE REVIEW

File: types/api.d.ts [hunk 1]
```diff
@@ -13,11 +13,12 @@ export type TokenDictionary = {
 export interface EvidenceRequestResponse
   extends Omit<
     IEvidenceRequest,
-    'createdAt' | 'deliveryMethods' | 'documentTypes'
+    'createdAt' | 'deliveryMethods' | 'documentTypes' | 'userRequestedBy'
```
Comment (neilmendum) [inline types/api.d.ts:16]: I don't think we need to include this in the `Omit` list. I think we only omit fields which are optional, or additional parsing needs to be done - for example `createdAt` we convert to `DateTime` in `response-mapper.ts`

File: src/components/ResidentTable.tsx [hunk 2]
```diff
@@ -65,6 +69,9 @@ export const ResidentTable: FunctionComponent<Props> = ({
             <td className="govuk-table__cell govuk-table__cell--numeric">
               {row.uploaded}
             </td>
+            <th scope="col" className="govuk-table__cell govuk-table__cell">
```
Comment (neilmendum) [inline src/components/ResidentTable.tsx:72]:
```suggestion
            <td className="govuk-table__cell">
```

File: src/components/ResidentTable.tsx [hunk 3]
```diff
@@ -65,6 +69,9 @@ export const ResidentTable: FunctionComponent<Props> = ({
...
+              {row.userRequestedBy}
+            </th>
```
Comment (neilmendum) [inline src/components/ResidentTable.tsx:74]:
```suggestion
            </td>
```

File: cypress/fixtures/evidence_requests/index.json [hunk 4]
```diff
@@ -39,7 +40,8 @@
       }
     ],
     "team": "Development Housing Team",
-    "reason": "staging-reason"
+    "reason": "staging-reason",
+    "userRequestedBy": "test@email"
```
Comment (neilmendum) [inline cypress/fixtures/evidence_requests/index.json:44]: Small thing but in other fixtures we used `test@hackney.gov.uk`. Could use the same for consistency

File: types/api.d.ts (hunk 1 above)
Comment (bogdan-zaharia) [reply types/api.d.ts:16]: 👍

File: src/components/ResidentTable.tsx (hunk 2 above)
Comment (bogdan-zaharia) [reply src/components/ResidentTable.tsx:72]: addressed

File: src/components/ResidentTable.tsx (hunk 3 above)
Comment (bogdan-zaharia) [reply src/components/ResidentTable.tsx:74]: addressed

File: cypress/fixtures/evidence_requests/index.json (hunk 4 above)
Comment (bogdan-zaharia) [reply cypress/fixtures/evidence_requests/index.json:44]: Sure, I'll change it to that

**Review by neilmendum — APPROVED** [review]

**Pull request merged**